### ISSUE TITLE
Custom Solidity errors are not passed correctly

### ISSUE THREAD

**Issue by CedarMist** [oasisprotocol/oasis-web3-gateway#429]
Using custom `error` type with `revert` in Solidity isn't passed correctly to the client.

```solidity
    error Blah(uint);
    function testRevert () external view {
        revert Blah(123);
    }
```

The Ethers v5 TypeChain provider will automatically decode errors, the code should return this as the `data`:

    0x966682ee000000000000000000000000000000000000000000000000000000000000007b

However, with Sapphire 0.6.2-testnet, it returns this instead:

    0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000306c6d6143376741414141414141414141414141414141414141414141414141414141414141414141414141414141423700000000000000000000000000000000

This is due to the error handler which re-encodes the errors:

https://github.com/oasisprotocol/oasis-web3-gateway/blob/285aefcffb7e65977f0d57fc9906c8d7fce6b95a/rpc/eth/api.go#L368-L382

The web3-gateway log for this is:

```json
{"caller":"grpc.go:242","err":"reverted: lmaC7gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB7","level":"error","method":"/oasis-core.RuntimeClient/Query","module":"grpc/client","msg":"request failed","req_seq":310310,"rsp":{"data":null},"ts":"2023-09-08T02:37:08.654508996Z"}
```

Which is handled by: 

https://github.com/oasisprotocol/oasis-web3-gateway/blob/285aefcffb7e65977f0d57fc9906c8d7fce6b95a/rpc/eth/api.go#L440-L446

Ethers v5 predictably decodes this correctly as: 

```
Error: call revert exception; VM Exception while processing transaction: reverted with reason string "lmaC7gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB7" [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="testRevert()", data="0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000306c6d6143376741414141414141414141414141414141414141414141414141414141414141414141414141414141423700000000000000000000000000000000", errorArgs=["lmaC7gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB7"], errorName="Error", errorSignature="Error(string)", reason="lmaC7gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAB7", code=CALL_EXCEPTION, version=abi/5.7.0)
```

**Comment by CedarMist** [comment]
If we look at the implementation in oasis-sdk:

https://github.com/oasisprotocol/oasis-sdk/blob/957380713924e01aed725c020579f7b3cd049786/runtime-sdk/modules/evm/src/lib.rs#L222-L241

If it doesn't start with the `Error(string)` ABI selector, (`0x08c379a0`) a maximum of 1024 bytes will be base64 encoded.

So, the runtime-sdk EVM doesn't differentiate between error's that it's parsed and encoded, versus raw errors... Which makes it problematic for us!

**Comment by CedarMist** [comment]
With the fix for oasis-sdk#1478 the web3 gateway should detect if the version is below 0.6.3, and use old behavior.

Once everything is upgraded the special case in web3 gateway can be removed.

**Comment by kostko** [comment]
Below or equal to 0.6.2 as the fix will be in 0.6.3.
